### PR TITLE
RCAL-484: Update datamodels for reference pixel correction

### PR DESCRIPTION
**Pull request by WilliamJamieson**
Resolves [RCAL-484](https://jira.stsci.edu/browse/RCAL-484)

This is the matching PR to spacetelescope/rad#270, in that it implements the updates to `roman_datamodels` to support adding the reference pixel correction's reference file.

**Checklist**
- [ ] Added entry in `CHANGES.rst` under the corresponding subsection
- [ ] updated relevant tests
- [ ] updated relevant documentation
- [ ] Passed romancal regression testing on Jenkins / PLWishMaster. Link: https://plwishmaster.stsci.edu:8081/job/RT/job/romancal/XXX/

### CODE REVIEW

File: src/roman_datamodels/random_utils.py [hunk 1]
```diff
@@ -138,6 +138,69 @@ def generate_array_float32(size=(4096, 4096), min=None, max=None, units=None, se
...
+
+    min: tuple (or None)
+        Min value for the sample axis.
+            - If None and shape is "square", then (-1.0, -1.0).
+            - If None and shape is "disk", then (0.0, 0.0).
+
+    max: tuple (or None)
+        Max value for the sample axis.
+            - If None and shape is "square", then (1.0, 1.0).
+            - If None and shape is "disk", then (1.0, 2pi).
```
Comment (PaulHuwe) [inline src/roman_datamodels/random_utils.py:158]: This says 0 to 2pi, but the code goes from -pi to pi.

**Review by PaulHuwe — CHANGES_REQUESTED** [review]
Can you add a maker utility and associated tests?

File: src/roman_datamodels/random_utils.py (hunk 1 above)
Comment (WilliamJamieson) [reply src/roman_datamodels/random_utils.py:158]: This does not matter. It's randomly selecting the arg of the complex number. You will get identical results.

**Comment by WilliamJamieson** [comment]
Note this PR has been rebased on #193 to prove that the `maker_utils` are properly implemented.

**Comment by PaulHuwe** [comment]
Can you add plwishmaster link, please

**Comment by PaulHuwe** [comment]
> Can you add a maker utility and associated tests?

LGTM otherwise.

**Review by PaulHuwe — APPROVED** [review]
LGTM

**Review by nden — APPROVED** [review]
LGTM.
Needs a rebase

**Review by schlafly — APPROVED** [review]

**Pull request merged**